### ISSUE TITLE
Inconsistent Tracking

### ISSUE THREAD

**Issue by ikaro88** [google-ar/arcore-unity-sdk#377]
hi, i have build the classic helloAR of arcore unity sdk, but after i have scanned the flor, if i put on it an object its position is inconsistent how you can see in the video.
https://www.youtube.com/watch?v=AYGU6afFLvo

is normal? there is a fix?

in this way i cant make my project

**Comment by pablisho** [comment]
Hi, what you show in the video is expected behaviour. The only way to help with this is to have an environment with more visual features (i.e. less white walls and floor, more objects, etc.). Let me know if this answers your question.

**Comment by ikaro88** [comment]
@pablisho  you mean that this is normal? Because in this way i can not make my projects...
if you mean that is a normal and not a bug....i suppose yes, that answers to my question...

**Comment by ikaro88** [comment]
@pablisho  btw today i try also outside with more fisual features and there is the same problem....
there is no solution?

**Comment by ikaro88** [comment]
by the way i tried also outside, inside with different things around....there are always this problem...

**Comment by small-potatoes** [comment]
Hi ikaro88, I am not sure if this is the official HelloAR example we ship in the ARCore Unity SDK.  Can I confirm that you are placing each Andy Model as a child of an anchor?  If not, you must use an anchor to maintain object pose relative to the real world.  If so, then the plane height may be adjusting and dragging the Andy's with it.

**Comment by ikaro88** [comment]
Yes, i checked that EVERY object is under an anchor, the problem is the anchor itself.
This is confimed by the fact that also the plane shifted when i played

**Comment by small-potatoes** [comment]
I will file an internal bug for this to assess if this is a device-specific tracking issue.  Can you please provide your device fingerprint (adb shell getprop ro.build.fingerprint) and confirm that this happens with the **unmodified helloar example** in the sdk?

**Comment by ikaro88** [comment]
I confirm that is a unmodified helloar example, and my device is a Samsung Galaxy A6.

Device Figerprint: "samsung/a6ltexx/a6lte:8.0.0/R16NW/A600FNXXU2ARG5:user/release-keys"
Tell me if you need something else.

So, can you help me? because otherwise i can not do my apps with arcore

**Comment by small-potatoes** [comment]
Hi @ikaro88 , it is possible that the drift is happening due to adjustments being made to the plane's height.  As an experiment, you may try creating a world anchor using [Session.CreateAnchor(hitPose, null)](https://developers.google.com/ar/reference/unity/class/GoogleARCore/Session#classGoogleARCore_1_1Session_1ae542e2d597fdde597c2ac355601250cf)) to see if the anchors still drift.  Please let me know the result of your experiment.

In the meantime, we will look into whether there is a device-specific issue; it will take some time to come to a resolution if this is the case but we will keep this thread updated.

**Comment by ikaro88** [comment]
I tried but the problem persist...

**Comment by ikaro88** [comment]
@small-potatoes  i tried just now the new version and still not work, the drift still persist :S

and if i replace
var anchor = hit.Trackable.CreateAnchor(hit.Pose);
with
var anchor = Session.CreateAnchor(hit.Pose, null);
the result is even worse :(

any help?

**Comment by small-potatoes** [comment]
Hi @ikaro88, I am sorry you continue to have issues with this; it is being looked into as a possibly device-specific tracking bug.  I will update you as soon as we have something.

**Comment by FUBAR2077** [comment]
Hey @ikaro88 it's probably "depth perception" problem and "perspective problem". these androids are not really in exact point that you seem they to be - you should place them directly from above to have the most consistent placement.
